doc/data-warehouse-schema.md

```diff
@@ -0,0 +1,311 @@
+
+- [Data warehouse schema](#data-warehouse-schema)
+    - [ERD - schema](#erd---schema)
+        - [How to generate the schema](#how-to-generate-the-schema)
+    - [Introduction](#introduction)
+        - [Facts tables](#facts-tables)
+        - [Dimensional tables](#dimensional-tables)
+    - [Star Schema for GOV.UK](#star-schema-for-govuk)
+        - [Intro](#intro)
+            - [Questions to answer with the schema](#questions-to-answer-with-the-schema)
+            - [Patterns of all questions](#patterns-of-all-questions)
+        - [Design: Fact tables](#design-fact-tables)
+            - [Daily metrics: table `facts_metrics`](#daily-metrics-table-factsmetrics)
+            - [Facts Editions: table `facts_editions`](#facts-editions-table-factseditions)
+        - [Dimensions](#dimensions)
+            - [Editions dimension](#editions-dimension)
+            - [Dates dimension](#dates-dimension)
+        - [Aggregations](#aggregations)
+            - [Monthly aggregations](#monthly-aggregations)
+                - [Monthly metrics: table `monthly_metrics`](#monthly-metrics-table-monthlymetrics)
+                - [Month dimension: table `month_dimensions`](#month-dimension-table-monthdimensions)
+                - [Benefits](#benefits)
+                - [Drawbacks](#drawbacks)
+
+<!-- markdown-toc end -->
+
+# Data warehouse schema
+
+This document describes the database schema for the data warehouse. This is
+technical documentation so it is recommended to read first [what is a data
+wareouse and why GOV.UK uses one][data-warehouse-what-why] in order to get a
+better understanding of the problem that the data warehouse aims to solve.
+
+## ERD - schema ##
+
+The image below represents the schema for the data warehouse. There are not many
+tables but some of them are quite big. It is imporant to understand their
+associations and the type of data that each table stores.
+
+![data warehouse Database Schema][database-schema-image]
+
+### How to generate the schema ###
+
+The schema is generated using [Rails ERD][rails-erd] running the following
+command:
+
+```bash
+$ bundle exec erd --filetype=png --connected --title 'Schema' --filename doc/images/schema
+```
+
+*Developer note*: The schema might not be an exact representation of the latest
+version of the schema, but it should be close enough to get an understanding of
+it.
+
+## Introduction
+
+Data warehouses use a different design from standard operational databases. The
+latter are optimized to maintain strict accuracy of data by updating real-time
+information. Data warehouses, by contrast, are designed to give a long-range
+view of data over time. They trade off transaction volume and instead specialize
+in data aggregation.
+
+The data warehouse for GOV.UK stores data collected from other systems
+([Feedex][feedex], [Google Analytics][google analytics], ...) which are mostly
+backed by operational databases and stores this information using a `star
+schema` database design.
```

Comment:
The point being made by this paragraph needs to be clearer.

"which are mostly backed by operational databases" - I don't think the reader should have to worry about this and it just adds confusion. 

"stores this information using a `star schema` database design" - this is the first reference to `star schema`, which feels overlooked.

Maybe less confusing to separate the following points:
- The data warehouse uses a star schema and this is a brief description of what it is
- The data warehouse populates its tables used a variety of different sources.

Reply:
I had added references to `start schema` when we talk about the new database design, and split both points.